stack.yaml

```diff
@@ -1,4 +1,6 @@
-resolver: lts-14.8
+resolver: lts-20.25
 packages:
 - .
 extra-deps: []
+
+allow-newer: true
```

Comment:
:x: 
stack's implementation of `allow-newer` is very broken: if you build with `allow-newer: true`, and then you later _remove_ `allow-newer: true`, your cache is now poisoned, stack will continue to use packages previously-built with newer versions, and we lose the reproducibility advantage which stack claims to have over cabal.

Which packages need allow-newer though? you didn't list any extra-deps, and every package in the LTS is supposed to be compatible with every other package, no?

Reply:
Ah yea I meant to highlight this. Without setting that flag we get a lot of extra-deps, I'll try to clean it up tonight